tests/test_overscanCorrection.py

```diff
@@ -347,57 +349,177 @@ def test_parallelOverscanCorrection(self):
+                # still a residual ramp in this region.  The large
+                # delta allows the POLY fit to pass, which has sub-ADU
+                # differences.
+                self.assertAlmostEqual(exposureCopy.image.array[5][0],
+                                       0.5 * (exposureCopy.image.array[5][4]
+                                              + exposureCopy.image.array[5][5]), delta=0.3)
+
+    def test_bleedParallelOverscanCorrectionFailure(self):
+        """Expect that this should reduce the image variance with a full fit.
+        The default fitType of MEDIAN will reduce the median value.
+
+        This needs to operate on a RawMock() to have overscan data to use.
+
+        This adds a large artificial bleed to the overscan region,
+        which should be masked and patched with the median of the
+        other pixels.
+        """
+        exposure = self.makeExposure(addRamp=True, isTransposed=False)
+        detector = exposure.getDetector()
+        amp = detector.getAmplifiers()[0]
+
+        maskedImage = exposure.getMaskedImage()
+        overscanBleedBox = lsst.geom.Box2I(lsst.geom.Point2I(4, 10),
+                                           lsst.geom.Extent2I(2, 3))
+        overscanBleed = afwImage.MaskedImageF(maskedImage, overscanBleedBox)
+        overscanBleed.set(10000, 0x0, 1)  # This level is below the mask threshold.
+
+        statBefore = computeImageMedianAndStd(exposure.image[amp.getRawDataBBox()])
+
+        for fitType in ('MEDIAN', 'MEDIAN_PER_ROW'):
+            # We only test these three types as this should cover the
+            # scalar calculations, the generic vector calculations,
+            # and the specific C++ MEDIAN_PER_ROW case.
+            exposureCopy = exposure.clone()
+            config = ipIsr.IsrTask.ConfigClass()
+            config.overscan.doParallelOverscan = True
+            config.overscan.parallelOverscanMaskGrowSize = 1
+            # Ensure we don't mask anything
+            config.overscan.maxDeviation = 100000
+            config.overscan.fitType = fitType
+            isrTask = ipIsr.IsrTask(config=config)
+
+            # isrTask.overscan.maskParallelOverscan(exposureCopy, detector)
+            oscanResults = isrTask.overscan.run(exposureCopy, amp)
+
+            self.assertIsInstance(oscanResults, pipeBase.Struct)
+            if fitType == 'MEDIAN':
+                self.assertIsInstance(oscanResults.imageFit, float)
+                self.assertIsInstance(oscanResults.overscanFit, float)
+            else:
+                self.assertIsInstance(oscanResults.imageFit, np.ndarray)
+                self.assertIsInstance(oscanResults.overscanFit, np.ndarray)
+            self.assertIsInstance(oscanResults.overscanImage, afwImage.ExposureF)
+
+            statAfter = computeImageMedianAndStd(exposureCopy.image[amp.getRawDataBBox()])
+            self.assertLess(statAfter[0], statBefore[0])
+
```

Comment:
Finally, in this case the std goes crazy with the MEDIAN_PER_ROW case, which should be confirmed.